pkgs/development/interpreters/python/hooks/setuptools-build-hook.sh

```diff
@@ -31,6 +31,17 @@ setuptoolsShellHook() {
         mkdir -p "$tmp_path/@pythonSitePackages@"
         eval "@pythonInterpreter@ -m pip install -e . --prefix $tmp_path \
           --no-build-isolation >&2"
+
+        # setuptools no longer installs a site.py file since version 49. Using
+        # a sitecustomize.py file is a workaround for making nix-shell
+        # environments work again.
+        # See https://github.com/pypa/setuptools/issues/2612
+        cat >"$tmp_path/@pythonSitePackages@/sitecustomize.py" <<EOF
```

Comment:
Our interpreter comes with a `sitecustomize.py`. I suppose this one could take precedence and so it works. This may stop working if we put more logic in the main one though.

I think we could use the existing `sitecustomize.py` for this by just exporting

```
NIX_PYTHONPATH="$tmp_path/@pythonSitePackages@"
```